Assets/Mirror/Runtime/ClientScene.cs

```diff
@@ -951,22 +951,29 @@ static void DestroyObject(uint netId)
             {
                 localObject.OnStopClient();
 
-                if (!InvokeUnSpawnHandler(localObject.assetId, localObject.gameObject))
+                // user handling
+                if (InvokeUnSpawnHandler(localObject.assetId, localObject.gameObject))
                 {
-                    // default handling
-                    if (localObject.sceneId == 0)
-                    {
-                        Object.Destroy(localObject.gameObject);
-                    }
-                    else
-                    {
-                        // scene object.. disable it in scene instead of destroying
-                        localObject.gameObject.SetActive(false);
-                        spawnableObjects[localObject.sceneId] = localObject;
-                    }
+                    // reset object after user's handler
+                    localObject.Reset();
                 }
+                // default handling
+                else if (localObject.sceneId == 0)
+                {
+                    // dont call reset before destroy so that values are still set in OnDestroy
+                    Object.Destroy(localObject.gameObject);
+                }
+                else
+                {
+                    // scene object.. disable it in scene instead of destroying
```

Comment:
very nice PR. could you just put thise comment above 'else' line so it's consistent with the first 'else if', and then merge yourself please?